pkg/imgpkg/lockconfig/images_lock.go

```diff
@@ -72,7 +82,16 @@ func (c ImagesLock) AsBytes() ([]byte, error) {
 		return nil, fmt.Errorf("Validating images lock: %s", err)
 	}
 
-	bs, err := yaml.Marshal(c)
+	// Use the first location instead of the value present in Image
```

Comment:
not sure why we cannot continue Image here.

Reply:
When we need to rewrite the ImagesLock file we want to write the PrimaryLocation that will be Image if the image was not copied, or the bundle repository if the image was copied.

We stopped changing the value o Image on the ImagesLock file

Reply:
hmm, may be it's better to have a method that can generate ImagesLock with PrimaryLocations (fills in image key).  AsBytes() would just do plain marshal.

Reply:
hmm, is there a situation where you want to call `AsBytes` and you do not want `PrimaryLocation` to be the value in the Image field?
If there are cases where we need this I would agree with you and can create a function like `AsBytesWithPrimaryLocation` or something. If not, I think this conversation makes sense in the `AsBytes`.

Reply:
to me this is more about symmetry between unmarshaling and marshaling. eg if i create ImagesLock with some Images (each ImageRef filled with Image), i shouldnt have to use AddLocation.

Reply:
You do not have to because by default the Image field is the `PrimaryLocation` of a particular image until you execute an `AddLocation`.

The call to `PrimaryLocation` here is to abstract the concept of the best location to retrieve a particular image from.

Do you see any use case where you would call `AsBytes` and would not want the Image field to be equal to the `PrimaryLocation` to retrieve an image from?